### PR TITLE
Added an option for positioning icons (left or right)

### PR DESCRIPTION
**Pull request by macig**
### Added an option for positioning icons (left or right)

sample

`g:lightline#bufferline#reverse_buffer_icon`

If set to `1`, buffer icons are displayed on the right side of the file name. If `0`, it will be displayed on the left side. Default is `0`.

### CODE REVIEW

**Comment by macig** [comment]
g:lightline#bufferline#reverse_buffer_icon
If set to 1, buffer icons are displayed on the right side of the file name. If 0, it will be displayed on the left side. Default is 0.

File: autoload/lightline/bufferline.vim [hunk 1]
```diff
@@ -71,14 +72,16 @@ function! s:get_buffer_name(i, buffer, path)
       let l:name = pathshorten(l:name)
     endif
   endif
-  if s:enable_devicons == 1 && exists('*WebDevIconsGetFileTypeSymbol')
+  if exists('*WebDevIconsGetFileTypeSymbol') && s:enable_devicons == 1 && s:reverse_buffer_icons == 0
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]: Why did you move the `exists` check to the beginning of the if expression?

File: autoload/lightline/bufferline.vim [hunk 2]
```diff
@@ -71,14 +72,16 @@ function! s:get_buffer_name(i, buffer, path)
...
     let l:name = WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t')) . ' ' . l:name
+  elseif s:reverse_buffer_icons == 1
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]: If there is only this one condition in the if expression then it will evaluate to true even if devicons are not enabled or installed. Also it will only apply to devicons and not nerdfont icons or nvim_web_devicons.

File: autoload/lightline/bufferline.vim [hunk 3]
```diff
@@ -71,14 +72,16 @@ function! s:get_buffer_name(i, buffer, path)
...
+    let l:name = l:name . ' ' . WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t'))
   elseif s:enable_devicons == 1 && has('nvim-0.5') && exists('g:nvim_web_devicons')
     let l:name = v:lua._bufferline_get_icon(bufname(a:buffer)) . ' ' . l:name
   elseif s:enable_nerdfont == 1
     try
       let l:name = nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0) . ' ' . l:name
-    catch /^Vim\%((\a\+)\)\=:E117:/
+      catch /^Vim\%((\a\+)\)\=:E117:/
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]: This change should be reverted.

File: autoload/lightline/bufferline.vim [hunk 4]
```diff
@@ -16,6 +16,7 @@ let s:number_separator    = get(g:, 'lightline#bufferline#number_separator', ' '
 let s:ordinal_separator   = get(g:, 'lightline#bufferline#ordinal_separator', '')
 let s:unnamed             = get(g:, 'lightline#bufferline#unnamed', '*')
 let s:reverse_buffers     = get(g:, 'lightline#bufferline#reverse_buffers', 0)
+let s:reverse_buffer_icons     = get(g:, 'lightline#bufferline#reverse_buffer_icon', 0)
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]: There is an **s** missing for the global variable name (icon => icon**s**). I also adjusted the spacing.
```suggestion
let s:reverse_buffer_icons = get(g:, 'lightline#bufferline#reverse_buffer_icons', 0)
```

File: autoload/lightline/bufferline.vim [hunk 5]
```diff
@@ -71,14 +72,16 @@ function! s:get_buffer_name(i, buffer, path)
...
+    let l:name = l:name . ' ' . WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t'))
   elseif s:enable_devicons == 1 && has('nvim-0.5') && exists('g:nvim_web_devicons')
     let l:name = v:lua._bufferline_get_icon(bufname(a:buffer)) . ' ' . l:name
   elseif s:enable_nerdfont == 1
     try
       let l:name = nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0) . ' ' . l:name
-    catch /^Vim\%((\a\+)\)\=:E117:/
+      catch /^Vim\%((\a\+)\)\=:E117:/
     endtry
   endif
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]: Your lines 75-86 should be replaced with this code which retrieves an icon and applies the positioning afterwards:
```viml
  let l:icon = s:get_icon(a:buffer)
  if l:icon != ''
    let l:name = s:reverse_buffer_icons == 0 ? (l:icon . ' ' . l:name) : (l:name . ' ' . l:icon)
  endif
```
The icon fetching code should be refactored into a new function:
```viml
function! s:get_icon(buffer)
  if s:enable_devicons == 1 && exists('*WebDevIconsGetFileTypeSymbol')
    return WebDevIconsGetFileTypeSymbol(fnamemodify(bufname(a:buffer), ':t'))
  endif
  
  if s:enable_devicons == 1 && has('nvim-0.5') && exists('g:nvim_web_devicons')
    return v:lua._bufferline_get_icon(bufname(a:buffer))
  endif

 if s:enable_nerdfont == 1
    try
      return nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0)
    catch /^Vim\%((\a\+)\)\=:E117:/
    endtry
  endif

  return ''
endfunction
```

This way the reverse buffer icons functionality works for all icon integrations.

**Review by mengelbrecht — CHANGES_REQUESTED** [review]

File: autoload/lightline/bufferline.vim (hunk 5 above)
Comment (macig) [reply autoload/lightline/bufferline.vim]: Thanks so much for pointing out my mistakes and for explaining in detail how to fix them.

File: README.md [hunk 6]
```diff
@@ -173,6 +173,11 @@ Default is `0` (no auto-show behavior).
 If enabled the buffers will be displayed in a reversed order.
 Default is `0` (buffers are not reversed).
 
+##### `g:lightline#bufferline#reverse_buffer_icons`
+
+If set to `1`, buffer icons are displayed on the right side of the file name.
+If `0`, it will be displayed on the left side. Default is `0`.
+
```
Comment (mengelbrecht) [inline README.md]:
```suggestion
##### `g:lightline#bufferline#icon_position`

 If set to `right`, filetype icons are displayed on the right side of the file name.
 If `left` or omitted, it will be displayed on the left side. Default is `left`.
 
```

File: autoload/lightline/bufferline.vim [hunk 7]
```diff
@@ -4,37 +4,38 @@
...
 endif
 if exists('g:lightline.component_raw.buffers')
-  let s:component_is_raw  = g:lightline.component_raw.buffers
+  let s:component_is_raw   = g:lightline.component_raw.buffers
 else
-  let s:component_is_raw  = 0
+  let s:component_is_raw   = 0
 endif
-let s:clickable           = has('tablineat') && s:component_is_raw ? get(g:, 'lightline#bufferline#clickable', 0) : 0
+let s:clickable            = has('tablineat') && s:component_is_raw ? get(g:, 'lightline#bufferline#clickable', 0) : 0
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]: Please revert this changes and instead add the following line below the definition of the `s:enable_nerdfont` variable:
```viml
let s:icon_position       = get(g:, 'lightline#bufferline#icon_position', 'left')
```

File: autoload/lightline/bufferline.vim [hunk 8]
```diff
@@ -71,16 +72,12 @@ function! s:get_buffer_name(i, buffer, path)
...
-    let l:name = v:lua._bufferline_get_icon(bufname(a:buffer)) . ' ' . l:name
-  elseif s:enable_nerdfont == 1
-    try
-      let l:name = nerdfont#find(fnamemodify(bufname(a:buffer), ':t'), 0) . ' ' . l:name
-    catch /^Vim\%((\a\+)\)\=:E117:/
-    endtry
+
+  let l:icon = s:get_icon(a:buffer)
+  if l:icon != ''
+    let l:name = s:reverse_buffer_icons == 0 ? (l:icon . ' ' . l:name) : (l:name . ' ' . l:icon)
```
Comment (mengelbrecht) [inline autoload/lightline/bufferline.vim]:
```suggestion
    let l:name = s:icon_position ==? 'right' ?  (l:name . ' ' . l:icon) : (l:icon . ' ' . l:name)
```

**Review by mengelbrecht — CHANGES_REQUESTED** [review]
Thanks for the PR @macig.

I thought about the naming of the setting and find `icon_position` clearer. Also the name is shorter which makes the spacing changes for the variable definitions obsolete.

**Comment by mengelbrecht** [comment]
Thanks for your PR @macig

**Comment by macig** [comment]
Thank you for the awesome plugin and for helping me implement my proposal.
